work_queue/src/work_queue_process.c

```diff
@@ -256,6 +259,81 @@ pid_t work_queue_process_execute(struct work_queue_process *p )
 		if(result == -1)
 			fatal("could not dup pipe to stderr: %s", strerror(errno));
 
+		if(strcmp(p->task->command_line, "@FUNCTION") == 0) {	
+			// read the contents from the input file
```

Comment:
You are doing all the right things here, but factor this out into a bunch of small functions, each one doing one little thing, something like this:

```
input = load_input_file( task )
output = invoke_coprocess_function( task, input )
write_output_file( output )
```

etc...